test/multidc/inter_dc_zmq_SUITE.erl

```diff
@@ -0,0 +1,453 @@
+    ok = gen_server:call(Pid, {publish, Message3}),
+
+    receive finish -> ok after 100 -> throw(subscriber_timeout) end,
+    gen_server:stop(Pid),
+    ok.
+
+pub_sub_stop(_Config) ->
+    application:set_env(antidote, pubsub_port, 15555),
+    {ok, 15555} = application:get_env(antidote, pubsub_port),
+    {ok, Pid} = inter_dc_pub:start_link(),
+
+    Message = <<"01", "ping">>,
+    Self = self(),
+
+    % start local subscriber
+    _Loop = spawn(fun() ->
+        {ok, SubSocket} = chumak:socket(sub),
+        chumak:subscribe(SubSocket, "01"),
+        {ok, SubPid} = chumak:connect(SubSocket, tcp, "localhost", 15555),
+        Self ! {sock, SubSocket, SubPid},
+        {ok, M} = chumak:recv(SubSocket),
+        ct:log("Received but should not receive something! ~p", [M]),
+        0 = M
+                  end),
+
+    {Subscriber, _SubscriberPid} = receive {sock, Sub, SubPid} -> {Sub, SubPid} end,
+
+    ct:log("Stopping subscriber"),
+
+    inter_dc_utils:close_socket(Subscriber),
+
+    timer:sleep(50),
+
+    % internal API, broadcast allows only for transactions
+    ok = gen_server:call(Pid, {publish, Message}),
+
+    gen_server:stop(Pid),
+    timer:sleep(50),
+    ok.
+
+pub_one_sub_two_topics_partitions(_Config) ->
+    application:set_env(antidote, pubsub_port, 15551),
+    {ok, 15551} = application:get_env(antidote, pubsub_port),
+    {ok, Pid} = inter_dc_pub:start_link(),
+
+    P1 = 0,
+    P2 = 365375409332725729550921208179070754913983135744,
+    P3 = 730750818665451459101842416358141509827966271488,
+    P4 = 1096126227998177188652763624537212264741949407232,
+
+    P1Bin = inter_dc_txn:partition_to_bin(P1),
+    P2Bin = inter_dc_txn:partition_to_bin(P2),
+    P3Bin = inter_dc_txn:partition_to_bin(P3),
+    P4Bin = inter_dc_txn:partition_to_bin(P4),
+
+    Msg = erlang:term_to_binary(helo),
+    M0 = <<P1Bin/binary, Msg/binary>>,
+    M1 = <<P1Bin/binary, Msg/binary>>,
+    M2 = <<P2Bin/binary, Msg/binary>>,
+    M3 = <<P3Bin/binary, Msg/binary>>,
+    M4 = <<P4Bin/binary, Msg/binary>>,
+
+    Self = self(),
+
+    % start local subscriber
+    spawn(fun() ->
+        {ok, SubSocket} = chumak:socket(sub),
+        chumak:subscribe(SubSocket, <<>>),
+        {ok, _SocketPid} = chumak:connect(SubSocket, tcp, "localhost", 15551),
+        {ok, MM1} = chumak:recv(SubSocket),
+        ct:log("Received test message ~p", [MM1]),
+
+
+        {ok, SubSocket2} = chumak:socket(sub),
+        ct:log("Socket 1: ~p   Socket 2: ~p",[SubSocket, SubSocket2]),
+
+        ct:log("Subscribing to ~p",[P1Bin]),
+        chumak:subscribe(SubSocket2, P1Bin),
+
+        ct:log("Subscribing to ~p",[P3Bin]),
+        chumak:subscribe(SubSocket2, P3Bin),
+
+        {ok, _} = chumak:connect(SubSocket2, tcp, "localhost", 15551),
+        timer:sleep(50),
+        Self ! more,
+
+        {ok, _MMM1} = chumak:recv(SubSocket2),
+        {ok, _MMM2} = chumak:recv(SubSocket2),
+
+        %% subscriber_timeout!
```

Comment:
Clean up?

Reply:
The subscriber_timeout comment is just a documentation on when the timeout on the chumak side happens, if you uncomment that line of code